projects/igniteui-angular/src/lib/grids/grid-base.directive.ts

```diff
@@ -6441,6 +6526,26 @@ export class IgxGridBaseDirective extends DisplayDensityBase implements
         }
     }
 
+    public endAdd(commit = true, event?: Event) {
+        const row = this.crudService.row;
+        const cell = this.crudService.cell;
+        if (!row && !cell) {
+            return;
+        }
+        if (commit) {
+            this.gridAPI.submit_add_value();
+            this.data.push(row.data);
```

Comment:
This does not take into consideration the transaction service. Should probably use  this.gridAPI.addRowToData instead. Related method and transaction pipe should probably also be updated to handle null PK and use the row instance instead. Otherwise add row will not work with transaction service enabled.